bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -108,184 +94,177 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 gatewayCommand = GatewayCommand.parse(code, command.toFullString());
             }
 
-            if (conn != null && conn.isConnected()) {
-                conn.sendCommand(gatewayCommand);
-
-                if (GatewayCommandCode.ControlSetpoint.equals(code)) {
-                    if (gatewayCommand.getMessage().equals("0.0")) {
-                        updateState(OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_CENTRAL_HEATING_WATER_SETPOINT,
-                                UnDefType.UNDEF);
-                    }
-                    updateState(OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_CENTRAL_HEATING_ENABLED,
-                            OnOffType.from(!gatewayCommand.getMessage().equals("0.0")));
-                } else if (GatewayCommandCode.ControlSetpoint2.equals(code)) {
-                    if (gatewayCommand.getMessage().equals("0.0")) {
-                        updateState(OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_CENTRAL_HEATING2_WATER_SETPOINT,
-                                UnDefType.UNDEF);
-                    }
-                    updateState(OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_CENTRAL_HEATING2_ENABLED,
-                            OnOffType.from(!gatewayCommand.getMessage().equals("0.0")));
+            sendCommand(gatewayCommand);
+
+            if (code.equals(GatewayCommandCode.CONTROLSETPOINT)) {
+                if (gatewayCommand.getMessage().equals("0.0")) {
+                    updateState(CHANNEL_OVERRIDE_CENTRAL_HEATING_WATER_SETPOINT, UnDefType.UNDEF);
                 }
-            } else {
-                connect();
```

Comment:
Just so you know: I fully agree with your deletion of the call to 'connect()` .. it was completely wrong to call that at this place :)